sdk/src/assertions/soft_binding.rs

```diff
@@ -0,0 +1,167 @@
+use serde::{Deserialize, Serialize};
+
+use super::labels;
+use crate::{
+    assertion::{Assertion, AssertionBase, AssertionCbor},
+    assertions::region_of_interest::RegionOfInterest,
+    cbor_types::UriT,
+    Result,
+};
+
+const ASSERTION_CREATION_VERSION: usize = 1;
+
+/// The data structure used to store one or more soft bindings across some or all of the asset's content.
+///
+/// https://c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#soft_binding_assertion
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq)]
+pub struct SoftBinding {
+    /// A string identifying the soft binding algorithm and version of that algorithm used to compute the value,
+    /// taken from the [C2PA soft binding algorithm list](https://github.com/c2pa-org/softbinding-algorithm-list).
+    ///
+    /// If this field is absent, the algorithm is taken from the `alg_soft` value of the enclosing structure.
+    /// If both are present, the field in this structure is used. If no value is present in any of these places,
+    /// this structure is invalid; there is no default.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub alg: Option<String>,
+
+    /// A list of details about the soft binding.
+    pub blocks: Vec<SoftBindingBlockMap>,
+
+    /// A human-readable description of what this hash covers.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+
+    /// A string describing parameters of the soft binding algorithm.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub alg_params: Option<String>,
+
+    /// Zero-filled bytes used for filling up space.
```

Comment:
We need to work on this. If we expect a user to add these fields correctly we need to know how do that. Or we can document that the SDK does this magically, but then the SDK needs to do that.

Reply:
From my understanding this field has no purpose or use outside of the lower level APIs. I'm inclined to private the fields, add a getter, and skip it from serialization although I'd like to hear your thoughts about that. These padding fields exist in a few assertions so we may want to do the same there.

Reply:
I moved the padding fields to methods since they aren't used in the SDK at a lower level. I've added additional documentation although I'm waiting to confirm with @mauricefisher64 if they are correct.